valves/polars.py

```diff
@@ -62,3 +62,44 @@ def bayes_average(
             / (C + pl.col(target_col).count().over(group_cols))
         ).alias(out_col)
     )
+
+
+def item_item_counts(dataf, user_col="user", item_col="item", attach=False):
+    """
+    Computers item-item overlap counts from user-item interactions, useful for recommendations.
+
+    This function is meant to be used in a `.pipe()`-line.
+
+    Arguments:
+        - dataf: polars dataframe
+        - user_col: name of the column containing the user id
+        - item_col: name of the column containing the item id
+    """
+    result = (
+        dataf.with_columns(
+            [
+                pl.col(pl.col(item_col))
+                .list()
+                .over("user")
+                .explode()
+                .alias("item_rec"),
+            ]
+        )
+        .filter(pl.col(item_col) != pl.col("item_rec"))
+        .with_columns(
+            [
+                pl.col(user_col).count().over(pl.col(item_col)).alias("n_item"),
+                pl.col(user_col).count().over("item_rec").alias("n_item_rec"),
```

Comment:
Side note:

This is not yet utilized. But I plan to built an optimization in polars. When we detect that we `over` the same groups multiple times, we can rewrite the query so that the groupby operation is only done once.

We had this under a mutex first, but this leads to a deadlock in the parallelization framework we use, and is not easily solvable. :(

https://github.com/rayon-rs/rayon/issues/592

Reply:
Good to know. The benchmarks in the future should show this. 

Either way, this is *much* faster than the `.transform()` stuff seen sofar.